test/functional/aws_test.go

```diff
@@ -0,0 +1,242 @@
+package functional
+
+import (
+	goctx "context"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3control"
+	"github.com/integr8ly/integreatly-operator/test/common"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"os"
+	"strings"
+	"testing"
+)
+
+var (
+	expectedRDSResources = []*rds.DBInstance{
+		{
+			DBName:             aws.String(ThreeScaleResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+		{
+			DBName:             aws.String(UPSResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+		{
+			DBName:             aws.String(CRResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+		{
+			DBName:             aws.String(CRResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+		//TODO implement Fuse here when it's ready
+		{
+			DBName:             aws.String(ClusterSSOResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+		{
+			DBName:             aws.String(ApplicationSSOResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+	}
+	expectedElasticacheResources = []*elasticache.ReplicationGroup{
```

Comment:
Can we add check for multi-az and automatic backup?

Reply:
ReplicationGroups by themselves don't have multi-AZ fields, this means that a slightly more extensive test is needed to check if the cache clusters for the respective groups are in different AZ's, but these should also be implemented, Noted to be added